lib/dash/dash_parser.js

```diff
@@ -334,16 +336,36 @@ shaka.dash.DashParser = class {
    * @private
    */
   async parseManifest_(data, finalManifestUri) {
-    const Error = shaka.util.Error;
-    const MpdUtils = shaka.dash.MpdUtils;
-    const TXml = shaka.util.TXml;
-
-    const mpd = TXml.parseXml(data, 'MPD');
-
+    let manifestData = data;
+    const manifestPreprocessor = this.config_.dash.manifestPreprocessor;
+    if (manifestPreprocessor) {
+      shaka.Deprecate.deprecateFeature(5,
+          'manifest.dash.manifestPreprocessor configuration',
+          'Please Use manifest.dash.manifestPreprocessorTXml instead.');
+      const mpdElement =
+          shaka.util.XmlUtils.parseXml(manifestData, 'MPD');
+      if (!mpdElement) {
+        throw new shaka.util.Error(
+            shaka.util.Error.Severity.CRITICAL,
+            shaka.util.Error.Category.MANIFEST,
+            shaka.util.Error.Code.DASH_INVALID_XML,
+            finalManifestUri);
+      }
+      manifestPreprocessor(mpdElement);
+      manifestData = shaka.util.XmlUtils.toArrayBuffer(mpdElement);
+    }
+    const mpd = shaka.util.TXml.parseXml(manifestData, 'MPD');
```

Comment:
So our backward compatibility solution is to parse it twice?  I suppose it will only happen if someone is using this config, so most users won't see a penalty... Probably okay, but I'm unsure.

Reply:
OK, sounds good.